src/main/java/server/ServerHandler.java

```diff
@@ -44,29 +44,19 @@ public ServerHandler(Viewer viewer) {
         validator = new Validator(viewer);
 
         commandsFactory = Map.of(
-                "/list", cmds -> cmds.split(" ").length == 1 ? new CommandList(sessionSet) : new CommandErrorSyntax(),
-                "/help", cmds -> cmds.split(" ").length == 1 ? new CommandHelp() : new CommandErrorSyntax(),
-                "/rooms", cmds -> cmds.split(" ").length == 1 ? new CommandRooms(rooms) : new CommandErrorSyntax(),
-                "/join", cmds -> {
-                    String[] strings = cmds.split(" ");
-                    if (strings.length != 2) {
-                        return new CommandErrorSyntax();
-                    }
-                    String nameOfRoom = strings[1];
-                    if (!validator.validateRoomName(nameOfRoom)) {
-                        return new CommandErrorSyntax();
-                    }
+                "/list", args -> args.length == 1 ? new CommandList(sessionSet) : "Command '/list' cannot have any args.",
+                "/help", args -> args.length == 1 ? new CommandHelp() : "Command '/help' cannot have any args.",
+                "/rooms", args -> args.length == 1 ? new CommandRooms(rooms) : "Command '/rooms' cannot have any args.",
+                "/join", args -> {
+                    if (args.length != 2) return "Command '/join' must have syntax '/join [room_name]'.";
```

Comment:
You can throw an exception instread. It will be couth in place where the factory is invoked